### DISCUSSION TITLE
Updates

### DISCUSSION THREAD

**Discussion by Systemedic** [mealie-recipes/mealie#2571] [General]
Hi,
Will this project ever get net updates? The latest release is from october 2022 (almost a year ago). I see there is work done in the github. But no new releases or docker update. Is the project still in development or is it retired?

**Comment by michael-genson** [comment]
While the last major release was October of last year, the nightly version has been updated quite a lot. If you're looking to migrate to the latest and greatest, I recommend switching to nightly.

There will be a new release soon™, most likely a full 1.0.0 release, but there are a few todo items left first.

**Reply by Systemedic** [reply]
Hi! Thank you Michael. I did not know that. I will look into that and guess i'll switch to the nightly!

**Reply by michael-genson** [reply]
No problem :)

**Comment by tweek11** [comment]
Since the nightly is being updated more consistently, that is until the full 1.0.0 version release as mentioned above - Does anything need to be done or changed other than switching the following in the the docker-compose.yml file

image: hkotel/mealie:frontend-v1.0.0beta-5 
to 
image: hkotel/mealie:frontend-nightly

 image: hkotel/mealie:api-v1.0.0beta-5
to 
 image: hkotel/mealie:api-nightly

Also does the omni-nightly need to be added into the docker-compose as well? Or is that a separate build entirely?

**Reply by michael-genson** [reply]
If you're planning on upgrading to nightly you probably want to check out the new simplified build. We're getting rid of the split container/omni builds entirely:

https://github.com/mealie-recipes/mealie/blob/mealie-next/docs/docs/documentation/getting-started/installation/sqlite.md
https://github.com/mealie-recipes/mealie/blob/mealie-next/docs/docs/documentation/getting-started/installation/postgres.md

TL;DR:
- no more frontend container
- new image for the backend: `ghcr.io/mealie-recipes/mealie:nightly` (which is now the only container, unless you're using postgres)
- expose internal port 9000 instead of 3000 (since the frontend container doesn't exist anymore)

**Comment by fakethinkpad85** [comment]
Once released, will docker users be updated (using portainer for instance) to using the single container automatically?
Also, will the next actual release be Beta RC1 milestone or the final 1.0.0?

Any guide available to upgrade/migrate Beta 5 to single container Master?

Thanks in advance

**Reply by michael-genson** [reply]
I'm not sure how portainer works, but docker doesn't have any sort of automatic update system. So how/when you migrate to 1.0.0 will be up to you.

There will be more detailed documentation, but the basic migration process from the split beta 5 containers is:
1. Remove the frontend container entirely (for any env vars you have on the frontend: just move them to the backend)
2. Replace the backend image with the 1.0.0 release build (this doesn't exist yet, but if you want to move early to nightly, you can use `ghcr.io/mealie-recipes/mealie:nightly`)
3. Expose internal port 9000 and route traffic there (currently you're routing traffic to internal port 3000). Of course you can map any external port you'd like to that

The underlying database and application are the same, it's just a different deployment structure, so none of your data should change. You should save a backup just in case, though (especially if moving from beta 5, since a lot has changed since then)

**Reply by Systemedic** [reply]
What I did was the following;
Made a backup of all the recipes and stuff. Stopped the container en deleted both the containers. Then I created a new container with the single image and mapped this to the same folder as the old containers were mapped. After booting the container all my settings and recipes were still there.

**Comment by fakethinkpad85** [comment]
Appreciate the quick response, might give it a try but not really knowledgeable about docker containers. Noticed however both the Gui and Api had backup endpoints for creating and restoring but the Gui mentioned it not being cross version compatible is that the case for Beta 5 to current mealie-next?

**Reply by michael-genson** [reply]
As long as your volumes are mounted you _should_ just be able to update the image and it just works. If not you can always restore your backup

> the Gui mentioned it not being cross version compatible

[Importing a backup from beta 5 into nightly should work fine now](https://github.com/mealie-recipes/mealie/pull/2213)

**Comment by fakethinkpad85** [comment]
Perfect, thanks again @michael-genson

**Comment by fakethinkpad85** [comment]
Update, for anyone else doing this in the future, i just stopped the Stack in Portainer and created a new one with the same volume path as the old one using the script as seen in the dokumentation and it worked like a charm.

https://nightly.mealie.io/documentation/getting-started/installation/sqlite/
